build/push-okd4.sh

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env bash
+
+validateOC() {
+  OC_USER_NAME=$(oc whoami 2>/dev/null)
+  if [ ! $? -eq 0 ]; then
+    echo "You must login with 'oc login' or set KUBECONFIG env variable"
+    exit 1
+  fi
+}
+
+validateOC
+
+DOCKER_CERTS_FOLDER="/etc/docker/certs.d/"
+REGISTRY_ROUTE=$(oc get route default-route -n openshift-image-registry -o json | jq -r '.spec.host')
+OC_PROJECT_NAME=$(oc project -q)
+PROJECT_NAME=${PROJECT_NAME-$OC_PROJECT_NAME}
+OC_USER_TOKEN=$(oc whoami --show-token)
+VERSION=$(git describe --tags --always --dirty)
+
+validateROOT() {
+  if ! [ $(id -u) = 0 ]; then
+    echo "It's nessesay to run next command with sudo!"
+    sudo ls >/dev/null
+    if [ ! $? -eq 0 ]; then
+      echo "Invalid sudo credentials"
+      exit 1
+    fi
+  fi
+}
+
+echo "$REGISTRY_ROUTE" "$PROJECT_NAME" "$OC_USER_NAME" "$OC_USER_TOKEN"
+
+if [ -d "${DOCKER_CERTS_FOLDER}${REGISTRY_ROUTE}" ]; then
+  echo "Instance folder detected"
+else
+  echo "Instance folder not found, creating..."
+  validateROOT
+  sudo -E mkdir -p "${DOCKER_CERTS_FOLDER}${REGISTRY_ROUTE}"
+fi
+
+if [ -f "${DOCKER_CERTS_FOLDER}${REGISTRY_ROUTE}/tls.crt" ]; then
+  echo "Instance file detected"
+else
+  echo "Instance file not found, extracting..."
+  validateROOT
+  sudo -E oc extract secret/router-ca --keys=tls.crt -n openshift-ingress-operator --to="${DOCKER_CERTS_FOLDER}${REGISTRY_ROUTE}"
+fi
+
+docker login -u "$OC_USER_NAME" -p "${OC_USER_TOKEN}" "${REGISTRY_ROUTE}"
```

Comment:
`oc registry login --skip-check` does the same job once you are logged in and you can drop the env vars.

Reply:
Thanks, @Crumby will update
It'a strange, that out KB note related to this problem, doesn't know about this option